Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript

```diff
@@ -5424,3 +5455,358 @@ private command EditorFieldCacheSwap pOldRuggedObjectId, pNewRuggedObjectId
+         if tMinOffset = 0 then 
+            put tOffset into tMinOffset
+         else
+            if tOffset < tMinOffset then 
+               put tOffset into tMinOffset
+            end if
+         end if
+      end if
+   end repeat
+   return tMinOffset
+end commentStartsAt
+
+command workOnBackSlash @tBackSlashText, @tOpenStructs, @tCounter
+   local tCommentStartChar
+   local tSubString
+   local tLastWord
+   
+   repeat for each line aLine in tBackSlashText
+      
+      if last char of aLIne is "\" then 
+         if word - 2 of aLine is "then" then
+            add 1 to tCounter
+            exit repeat
+         else
+            add 1 to tCounter
+            next repeat
+         end if
+      end if
+      if word 1 of aLine is "then" then
+         put commentStartsAt(aLine) into tCommentStartChar
+         if tCommentStartChar > 0 then
+            put char 1 to tCommentStartChar - 1 of aLine into tSubString
+            put word - 1 of tSubString into tLastWord
+         else
+            put word - 1 of aLine into tLastWord
+         end if
+         if tLastWord = "then" then
+            add 1 to tCounter
+            next repeat
+         else
+            --add 1 to tCounter
+            exit repeat
+         end if
+      end if
+      if word 1 to 2 of aLine = "end if" then
+         add 1 to tCounter
+         exit repeat
+      end if
+   end repeat
+end workOnBackSlash
+
+private command foldUnfold pStart, pEnd
+   lock screen
+   set the hidden of line pStart + 1 to pEnd  of field "Script" of me to not the hidden of line pStart +1 to pEnd of field "Script" of me
+   if the hidden of line pStart +1 of field "Numbers" of me then
+      set the textStyle["underline"] of line pStart to pEnd of field "Script" of me to false
+   else
+      set the textStyle["underline"] of line pStart of field "Script" of me to true
+   end if
+   set the hidden of line pStart + 1 to pEnd  of field "Numbers" of me to not the hidden of line pStart +1 to pEnd of field "Numbers" of me
+   unlock screen
+end foldUnfold
+
+on unfoldAll
+   lock screen
+   set the hidden of line  1 to -1  of field "Script" of me to false
+   set the hidden of line 1 to - 1  of field "Numbers" of me to false
+   set the textStyle["underline"] of line 1 to - 1 of field "Script" of me to false
+   unlock screen
+end unfoldAll
+
+on foldAll
+   local tHasHidden
+   put the hidden of line 1 to - 1 of field "Script" of me into tHasHidden
+   lock screen
+   -- unfold first is some lines are hidden
+   if tHasHidden is "mixed" or tHasHidden is true then
+      unfoldAll
+   end if
+   
+   local tHandlers
+   local tScriptHTML
+   local tNumbersHTML
+   local tScriptNewHTML
+   local tNumbersNewHTML
+   local tHandlerStart
+   local tHandlerEnd
+   local tHandlerLastEnd
+   local tHidden
+   local tBeginU
+   local tEndU
+   local tScriptTemp
+   local tNumbersTemp
+   local tUnderlineText
+   
+   put " hidden" into tHidden
+   put "<u>" into tBeginU
```

Comment:
I wonder if folding to something like this would be better than the underline thing:
```
command MyCommand
end MyCommand
```